contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/AbstractHttpHandler.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.http.core;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.opentelemetry.contrib.http.core.HttpTraceConstants.INSTRUMENTATION_LIB_ID;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.distributedcontext.DistributedContext;
+import io.opentelemetry.metrics.Meter;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Event;
+import io.opentelemetry.trace.Span;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/** Base class for handling request on http client and server. */
+abstract class AbstractHttpHandler<Q, P> {
+
+  /** The {@link HttpExtractor} used to extract information from request/response. */
+  @VisibleForTesting final HttpExtractor<Q, P> extractor;
+
+  @VisibleForTesting final StatusCodeConverter statusConverter;
+  private final Meter meter;
+
+  /** Constructor to allow access from same package subclasses only. */
+  AbstractHttpHandler(
+      HttpExtractor<Q, P> extractor, StatusCodeConverter statusConverter, Meter meter) {
+    checkNotNull(extractor, "extractor is required");
+    this.extractor = extractor;
+    if (statusConverter == null) {
+      this.statusConverter = new StatusCodeConverter();
+    } else {
+      this.statusConverter = statusConverter;
+    }
+    if (meter == null) {
+      this.meter = OpenTelemetry.getMeterFactory().get(INSTRUMENTATION_LIB_ID);
+    } else {
+      this.meter = meter;
+    }
+  }
+
+  /**
+   * A convenience to record a {@link Event} with given parameters.
+   *
+   * @param span the span which this {@code Event} will be added to.
+   * @param type the type of event.
+   * @param uncompressedMessageSize size of the message before compressed (optional).
```

Comment:
these parameters are not optional, since they are little-l longs. 0 seems like a totally valid message size, although I don't understand what "message" means in this context.